File: WorkSphere.Server/Model/Salary.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WorkSphere.Model;

namespace WorkSphere.Server.Model
{
    public class Salary : BaseEntity
    {
        [Required]
        [Precision(18, 2)]
        public decimal BasicSalary { get; set; }

        [Precision(18, 2)]
        public decimal? Bonus { get; set; }

        [Precision(18, 2)]
        public decimal? Deductions { get; set; }

        // Calculated Total Salary
        [NotMapped]
        public decimal TotalSalary => (BasicSalary + (Bonus ?? 0)) - (Deductions ?? 0);

        // One-to-One: Salary is tied to an Employee
        public int? EmployeeID { get; set; }

        // [JsonIgnore]
        public virtual Employee? Employee { get; set; }

        // One-to-One: Salary is tied to a Project Manager
        public int? ProjectManagerID { get; set; }

        // [JsonIgnore]
        public virtual ProjectManager? ProjectManager { get; set; }
    }
}

File: WorkSphere.Client/src/app/models/salary.ts
export class Salary {

  public id: number
  public basicSalary: number
  public bonus: number
  public deductions: number
  public totalSalary: number
  public employeeID: number

  constructor(id: number, basicSalary: number, bonus: number, deductions: number, totalSalary: number, employeeID: number) {
    this.id = id
    this.basicSalary = basicSalary
    this.bonus = bonus
    this.deductions = deductions
    this.totalSalary = totalSalary
    this.employeeID = employeeID
  }
}
